src/rendergraph/common/types.cpp

```diff
@@ -0,0 +1,111 @@
+#include "rendergraph/types.h"
+
+#include <QMatrix4x4>
+#include <QVector2D>
+#include <QVector3D>
+#include <QVector4D>
+#include <cstdint>
+
+using namespace rendergraph;
+
+int rendergraph::sizeOf(PrimitiveType type) {
+    switch (type) {
+    case PrimitiveType::UInt:
+        return sizeof(uint32_t);
+    case PrimitiveType::Float:
+        return sizeof(float);
+    }
+    return 0;
```

Comment:
Should we add a `DEBUG_ASSERT` here to detect potentially unsupported new renderers?